src/main/java/org/akaza/openclinica/control/admin/ScheduleJobServlet.java

```diff
@@ -0,0 +1,176 @@
+import core.org.akaza.openclinica.job.JobTriggerListener;
+import core.org.akaza.openclinica.job.OpenClinicaSchedulerFactoryBean;
+import core.org.akaza.openclinica.service.PermissionService;
+import core.org.akaza.openclinica.web.InsufficientPermissionException;
+import org.akaza.openclinica.control.SpringServletAccess;
+import org.akaza.openclinica.control.core.SecureController;
+import org.akaza.openclinica.control.form.FormProcessor;
+import org.akaza.openclinica.control.form.Validator;
+import org.akaza.openclinica.view.Page;
+import org.apache.commons.lang.StringUtils;
+import org.quartz.Scheduler;
+import org.quartz.TriggerKey;
+import org.quartz.impl.StdScheduler;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.sql.DataSource;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Properties;
+
+import static core.org.akaza.openclinica.dao.hibernate.multitenant.CurrentTenantIdentifierResolverImpl.CURRENT_TENANT_ID;
+
+public class ScheduleJobServlet extends SecureController {
+    protected static final String PERIOD = "periodToRun";
+    protected static final String FORMAT_ID = "formatId";
+    protected static final String DATASET_ID = "dsId";
+    protected static final String DATE_START_JOB = "job";
+    protected static final String EMAIL = "contactEmail";
+    protected static final String JOB_NAME = "jobName";
+    protected static final String JOB_DESC = "jobDesc";
+    protected static final String USER_ID = "user_id";
+    protected static String TRIGGER_IMPORT_GROUP = "importTrigger";
+    protected PermissionService permissionService;
+
+    @Override
+    protected void mayProceed() throws InsufficientPermissionException {
+        if (ub.isSysAdmin() || ub.isTechAdmin()) {
+            return;
+        }
+
+        addPageMessage(respage.getString("no_have_correct_privilege_current_study") + respage.getString("change_study_contact_sysadmin"));
+        throw new InsufficientPermissionException(Page.MENU_SERVLET, resexception.getString("not_allowed_access_extract_data_servlet"), "1");
+    }
+
+    protected StdScheduler getScheduler() {
+        scheduler = this.scheduler != null ? scheduler : (StdScheduler) SpringServletAccess.getApplicationContext(context).getBean(SCHEDULER);
+        return scheduler;
+    }
+
+    protected PermissionService getPermissionService() {
+        return permissionService = (PermissionService) SpringServletAccess.getApplicationContext(context).getBean("permissionService");
+    }
+
+    @Override
+    protected void processRequest() throws Exception {
```

Comment:
Instead of an empty implementation, make this an abstract method that the subclasses need to implement